### PR TITLE
Fix Mingw-w64 build

### PR DESCRIPTION
**Pull request by KindDragon**
@BillyDonahue PR #721 break build with Mingw-w64, this PR should fix this

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -1,71 +1,84 @@
...
-    {
-        $generator = "$generator Win64"
+    Write-Output "Compiler: $env:compiler"
+    Write-Output "Generator: $env:generator"
+
+    # git bash conflicts with MinGW makefiles
+    if ($env:generator -eq "MinGW Makefiles") {
+        $env:path = $env:path.replace("C:\Program Files\Git\usr\bin;", "")
+        if ($env:compiler -eq "gcc-5.3.0-posix") {
+            $cxx_path = "C:\mingw-w64\i686-5.3.0-posix-dwarf-rt_v4-rev0\mingw32\bin"
```
Comment (BillyDonahue) [inline appveyor.yml]: Why are we being so specific here and specifying a path off of C:\ ?

Sorry I don't understand why so many files are being rewritten here.

File: googletest/include/gtest/internal/gtest-port.h [hunk 2]
```diff
@@ -396,7 +400,7 @@
 #  include <io.h>
 # endif
 // In order to avoid having to include <windows.h>, use forward declaration
-#if GTEST_OS_WINDOWS_MINGW
+#if GTEST_OS_WINDOWS_MINGW && !(__MINGW64_VERSION_MAJOR >= 5)
```
Comment (BillyDonahue) [inline googletest/include/gtest/internal/gtest-port.h]: && (__MINGW64_VERSION_MAJOR < 5)

File: appveyor.yml (hunk 1 above)
Comment (KindDragon) [inline appveyor.yml]: AppVeyor has preinstalled mingw-w64 i686-5.3.0 https://github.com/appveyor/ci/issues/861

Comment (KindDragon) [inline appveyor.yml]: Sorry for bugging you again what with PR #723 and #856? I'm sorry if I'm too intrusive :blush:

File: googletest/include/gtest/internal/gtest-port.h (hunk 2 above)
Comment (KindDragon) [inline googletest/include/gtest/internal/gtest-port.h]: fixed

**Comment by KindDragon** [comment]
No it should works with all cases

**Comment by dneto0** [comment]
This patch works for me with MinGW 4.8.2.  I'm not exercising the AppVeyor part of of it though.

File: googletest/include/gtest/internal/gtest-port.h [hunk 3]
```diff
@@ -396,7 +396,7 @@
 #  include <io.h>
 # endif
 // In order to avoid having to include <windows.h>, use forward declaration
-#if GTEST_OS_WINDOWS_MINGW
+#if GTEST_OS_WINDOWS_MINGW && !defined(__MINGW64_VERSION_MAJOR)
```
Comment (dneto0) [inline googletest/include/gtest/internal/gtest-port.h]: I can confirm that MinGW 4.8.2 g++ does not define this symbol.
The only symbols defined by that compiler with "MINGW" in its name are:
* `__MINGW32__` for the 32 bit compiler.
* both `__MINGW32__` and `__MINGW64__` for the 64 bit compiler

**Comment by Mido** [comment]
This PR fixes google-test building for me. Today's (commit `ed9d1e1ff92ce199de5ca2667a667cd0a368482a`) google-test `master` branch doesn't work for me currently. Environment:

Windows 10
Mingw-w64: x86_64-6.1.0-win32-seh-rt_v5-rev0
CMake 3.6
CLion 2016.2.1

I'm looking forward to this PR :+1:

**Comment by dneto0** [comment]
Hi @BillyDonahue please take another look at this.
Currently I can't cross compile from Linux to Windows with the stock mingw-4.8.2 on Ubuntu trusty.  This patch fixes the problem.

**Comment by rakhimov** [comment]
Is this the old mingw/msys or the new msys2/mingw64 from https://github.com/Alexpux/MINGW-packages?

**Comment by dneto0** [comment]
@rakhimov  I'm using g++-mingw-w64 4.8.2-10ubuntu2+12 from Ubuntu trusty.  As far as I know it's the stock mingw for Ubuntu trusty.

**Pull request merged**